wiring/src/spark_wiring_interrupts.cpp

```diff
@@ -107,12 +102,9 @@ bool attachInterrupt(uint16_t pin, raw_interrupt_handler_t handler, InterruptMod
  *******************************************************************************/
 void detachInterrupt(uint16_t pin)
 {
-#if Wiring_Cellular == 1
-    /* safety check that prevents users from detaching an interrupt from
-     * BATT_INT_PC13 for power management which is shared with D7 */
-    if (pin == D7) return;
-#endif
-    HAL_Interrupts_Detach(pin);
+    if (SYSTEM_ERROR_NONE != HAL_Interrupts_Detach(pin)) {
+        return;
+    }
     delete handlers[pin];
```

Comment:
Let's also add an `if` check here:
suggestion```
if (handlers[pin]) {
    delete handlers[pin];
    handlers[pin] = nullptr;
```

Reply:
Fixed this line code for using`if` check